lib/utils/helpers.ts

```diff
@@ -356,3 +375,28 @@ export async function workaroundWindowsDnsIssue(ipOrHostname: string) {
 		await new Promise(r => setTimeout(r, delay));
 	}
 }
+
+/**
+ * Error handling wrapper around the npm `which` package:
+ * "Like the unix which utility. Finds the first instance of a specified
+ * executable in the PATH environment variable. Does not cache the results,
+ * so hash -r is not needed when the PATH changes."
+ *
+ * @param program Basename of a program, for example 'ssh'
+ * @returns The program's full path, e.g. 'C:\WINDOWS\System32\OpenSSH\ssh.EXE'
+ */
+export async function which(program: string): Promise<string> {
+	const whichMod = await import('which');
+	let programPath: string;
+	try {
+		programPath = await whichMod(program);
+	} catch (err) {
+		if (err.code === 'ENOENT') {
+			throw new ExpectedError(
+				`'${program}' program not found. Is it installed?`,
+			);
+		}
+		throw err;
+	}
+	return programPath;
+}
```

Comment:
The `which` function was added as it allows us to avoid using a shell to invoke 'ssh'. Avoiding a shell avoids having to deal with the incompatibilities between the Windows `cmd.exe` on and the Unix `/bin/sh`.